usr/src/uts/common/brand/lx/syscall/lx_socket.c

```diff
@@ -3974,6 +3974,21 @@ lx_getsockopt_socket(sonode_t *so, int optname, void *optval,
 	lx_socket_aux_data_t *sad;
 
 	switch (optname) {
+	case LX_SO_PROTOCOL:
+		/*
+		 * We need to special-case netlink.
+		 * XXX KEBE ASKS, maybe other non-INET/INET6 ones too?
```

Comment:
THIS ONE needs answering (and the comment needs removal).

Reply:
Overall this looks fine. I think you could probably not worry
about other families since we don't support much more in lx,
unless you have a specific example which you know is also
broken.

Reply:
If by "broken", you mean "LX behaves differently from stock Linux", then AF_UNIX also (mis)behaves.  See the test program I attached to the bug report.  Using 'p' demonstrates this.

Reply:
Sorry I wasn't clear, I meant a broken application. Unfortunately there are lots of subtle differences between Linux and lx but we usually only do "best effort" in lx unless is breaks an app.

Reply:
Added new version.  I think this will satisfy things.